src/client/Client.cc

```diff
@@ -1857,9 +1855,19 @@ int Client::make_request(MetaRequest *request,
       session = &mds_sessions.at(mds);
     }
 
+    if (request->async_dirop_cb)
+      (this->*request->async_dirop_cb)(request, session, 0);
+
```

Comment:
Some notes on the design would be nice here. This part is interesting in that a new callback has been added, which gets done _before_ the request is transmitted. Why was it done this way? It seems like we could do that just after the send is done?

The way we designed the kclient, for instance, we attempt to send the request as quickly as we can, and then just assume that the MDS will accept it and do the cleanup afterward.

All of the cleanup here is done before the send is done. Is that to reduce mutex thrashing? I don't get it.